tests/test_gh.py

```diff
@@ -0,0 +1,219 @@
+import pytest
+from unittest.mock import patch, MagicMock, Mock
+from gha_runner.gh import TokenRetrievalError, GitHubInstance
+from github import SelfHostedActionsRunner
+
+
+@pytest.fixture
+def github_release_mock():
+    # Create MagicMocks for components not directly patched
+    mock_release = MagicMock()
+    mock_asset = MagicMock()
+    mock_runners = MagicMock()
+    runner = MagicMock()
+
+    # Setup fixed attributes for mocks
+    asset_name = "runner-linux-x64.tar.gz"
+    asset_url = "https://github.com/testing/runner-linux-x64.tar.gz"
+    mock_asset.name = asset_name
+    mock_asset.browser_download_url = asset_url
+    runner.labels.return_value = [{"name": "runner-linux-x64"}]
+    mock_runners.__iter__.return_value = [runner]
+
+    # Using patch as a context manager inside the fixture
+    with patch("gha_runner.gh.Github") as mock_github:
+        mock_repo = MagicMock()
+        mock_github.return_value.get_repo.return_value = mock_repo
+        mock_repo.get_latest_release.return_value = mock_release
+        mock_repo.get_self_hosted_runners.return_value = mock_runners
+        mock_release.get_assets.return_value = [mock_asset]
+        mock_repo.remove_self_hosted_runner.return_value = True
+
+        instance = GitHubInstance("test", "testing/testing")
+
+        yield instance, mock_asset, mock_repo
+
+
+@pytest.mark.parametrize(
+    "platform, architecture, expected, raises, match_msg",
+    [
+        # Test case where the latest release exists
+        (
+            "linux",
+            "x64",
+            "https://github.com/testing/runner-linux-x64.tar.gz",
+            None,
+            None,
+        ),
+        # Test case where the latest release does not exist
+        ("darwin", "x64", None, RuntimeError, "Runner not found for darwin and x64"),
+    ],
+)
+def test_get_latest_runner_release(
+    github_release_mock, platform, architecture, expected, raises, match_msg
+):
+    instance, _, _ = github_release_mock
+    if raises:
+        with pytest.raises(raises, match=match_msg):
+            instance.get_latest_runner_release(platform, architecture)
+    else:
+        result = instance.get_latest_runner_release(platform, architecture)
+        assert result == expected, f"Expected URL {expected} but got {result}"
+
+
+@pytest.mark.parametrize(
+    "label, expected",
+    [
+        # Test case where the runner exists
+        ("runner-linux-x64", SelfHostedActionsRunner),
+        # Test case where the runner does not exist
+        ("runner-darwin-x64", None),
+    ],
+)
+def test_get_runner(github_release_mock, label, expected):
+    instance, _, _ = github_release_mock
+    result = instance.get_runner(label)
+    assert (result is not None) == (expected is not None)
```

Comment:
use `MagicMock`with `spec=SelfHostedActionsRunner` for `runner` in the mock. This will allow you to at least test as `isinstance` here.

```python
In [1]: from unittest.mock import MagicMock
   ...: class A:
   ...:     pass
   ...:
   ...: m = MagicMock(spec=A)

In [2]: isinstance(m, A)
Out[2]: True
```